projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SuggestionRecyclerAdapter.java

```diff
@@ -2,45 +2,76 @@
+
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.TextView;
 import android.widget.Toast;
 
+import java.util.ArrayList;
+
 import hasadna.noloan.common.FirestoreClient;
 import hasadna.noloan.common.SmsMessages;
 import hasadna.noloan.protobuf.SmsProto.SmsMessage;
 
-public class SuggetionRecyclerAdapter
-    extends RecyclerView.Adapter<SuggetionRecyclerAdapter.RecyclerViewHolder> {
+public class SuggestionRecyclerAdapter
+    extends RecyclerView.Adapter<SuggestionRecyclerAdapter.RecyclerViewHolder> {
+
+  ArrayList<SmsMessage> messages;
 
-  SmsMessages DbMessages;
+  public SuggestionRecyclerAdapter() {
+    messages = new ArrayList<>();
+    SmsMessages dbMessages = SmsMessages.get();
+
+    // maybe add getApproved to SmsMessages
+    for (SmsMessage message : dbMessages.getDbMessages()) {
+      if (!message.getApproved()) {
+        this.messages.add(message);
+      }
+    }
 
-  public SuggetionRecyclerAdapter() {
-    DbMessages = SmsMessages.get();
     Handler handler = new Handler(Looper.getMainLooper());
 
-    DbMessages.setMessagesListener(
-        new SmsMessages.MessagesListener() {
-          @Override
-          public void messageAdded(SmsMessage smsMessage) {
-            handler.post(() -> notifyItemInserted(DbMessages.getDbMessages().size()));
-          }
-
-          @Override
-          public void messageRemoved(int index, SmsMessage smsMessage) {
-            handler.post(() -> notifyItemRemoved(index));
-          }
-
-          @Override
-          public void messageModified(int index) {
-            handler.post(() -> notifyItemChanged(index));
-          }
-        });
+    dbMessages.setMessagesListener(new SmsMessages.MessagesListener() {
+      @Override
+      public void messageAdded(SmsMessage smsMessage) {
+        if (!smsMessage.getApproved()) {
+          messages.add(smsMessage);
+          handler.post(() -> notifyItemInserted(messages.size()));
+        }
+      }
+
+      @Override
+      public void messageRemoved(int index, SmsMessage smsMessage) {
+        if (!smsMessage.getApproved()) {
+          int i = SmsMessages.searchMessage(smsMessage, messages);
+          messages.remove(smsMessage);
+          handler.post(() -> notifyItemRemoved(i));
+        }
+      }
+
+      @Override
+      public void messageModified(int index) {
+        SmsMessage smsMessage = dbMessages.getDbMessages().get(index);
+        int i = SmsMessages.searchMessage(smsMessage, messages);
+
+        if (i != -1 && !smsMessage.getApproved()) {
+          messages.set(i, smsMessage);
+          handler.post(() -> notifyItemChanged(i));
+        } else if (i != -1) {
+          messages.remove(i);
+          handler.post(() -> notifyItemRemoved(i));
+        } else {
+          messages.add(smsMessage);
+          handler.post(() -> notifyItemInserted(messages.size()));
```

Comment:
Please add comments as you've done in `SpamRecyclerAdapter`, describing for which cases these conditions are here.